orc8r/cloud/go/services/analytics/collector_servicer.go

```diff
@@ -18,32 +18,127 @@ import (
+func (svc *CollectorService) filterResults(results []*protos.CalculationResult) []*protos.CalculationResult {
+	userStateMgr := svc.userStateMgr
+	// nothing to filter
+	if userStateMgr == nil {
+		return results
+	}
+
+	filteredResults := []*protos.CalculationResult{}
+	analyticsConfig := svc.analyticsConfig
+	for _, result := range results {
+		// enforce the minimum user threshold constraint at network level, gateway level and deployment level
+		if metricConfig, ok := analyticsConfig.Metrics[result.GetMetricName()]; ok {
+			if metricConfig.Export {
+				if metricConfig.EnforceMinUserThreshold {
+					networkID, networkLabelPresent := result.Labels[metrics.NetworkLabelName]
+					gatewayID, gatewayLabelPresent := result.Labels[metrics.GatewayLabelName]
+
+					if networkLabelPresent {
+						userNetworkTable := userStateMgr.UsersNetworkTable[networkID]
+						if gatewayLabelPresent {
+							if numUsers, ok := userNetworkTable.UsersGatewayTable[gatewayID]; ok {
+								if numUsers > analyticsConfig.MinUserThreshold {
+									filteredResults = append(filteredResults, result)
+								} else {
+									glog.V(1).Infof("metric %s(gateway label) dropped,active users %d below user threshold %d",
+										result.GetMetricName(), numUsers, analyticsConfig.MinUserThreshold)
+								}
+							}
+						} else {
+							if userNetworkTable.TotalUsersPerNetwork > analyticsConfig.MinUserThreshold {
+								filteredResults = append(filteredResults, result)
+							} else {
+								glog.V(1).Infof("metric %s(network label) dropped,active users %d below user threshold %d",
+									result.GetMetricName(), userNetworkTable.TotalUsersPerNetwork, analyticsConfig.MinUserThreshold)
+							}
+						}
+					} else {
+						if userStateMgr.TotalUsers > analyticsConfig.MinUserThreshold {
+							filteredResults = append(filteredResults, result)
+						} else {
+							glog.V(1).Infof("metric %s(deployment) dropped, active users %d below user threshold %d", result.GetMetricName(),
+								userStateMgr.TotalUsers, analyticsConfig.MinUserThreshold)
+						}
+					}
+				} else {
+					filteredResults = append(filteredResults, result)
+				}
+			} else {
+				glog.V(1).Infof("Metric %s export configuration disabled", result.GetMetricName())
+			}
+		} else {
+			glog.Errorf("Metric Configuration not found for %s", result.GetMetricName())
+		}
+	}
+	return filteredResults
+}
+
+// registerResults registers the computed metric with Prometheus
+func (svc *CollectorService) registerResults(calcParams *calculations.CalculationParams, results []*protos.CalculationResult) {
+	filteredResults := []*protos.CalculationResult{}
+	analyticsConfig := svc.analyticsConfig
+	for _, result := range results {
+		if metricConfig, ok := analyticsConfig.Metrics[result.GetMetricName()]; ok {
+			if metricConfig.Register {
+				filteredResults = append(filteredResults, result)
+			}
+		} else {
+			glog.Errorf("Metric Configuration not found for %s", result.GetMetricName())
```

Comment:
Configuration should be lowercase

Also, for clarity in the logs, consider adding single quotes around the `%s`